test/functional/tools/select_from_dataset.xml

```diff
@@ -57,6 +57,26 @@ echo select_collection '$select_collection' >> '$output'
                 </assert_contents>
             </output>
         </test>
+        <!-- test that dynamic selects can be chosen by name -->
+        <test>
+            <param name="single" value="2.tabular" />
+            <param name="select_single" value="68" />
+            <param name="multiple" value="2.tabular" />
+            <param name="select_multiple" value="71,62" />
+            <param name="collection">
+                <collection type="list">
+                    <element name="2" value="2.tabular" />
+                </collection>
+            </param>
+            <param name="select_collection" value="75,58" />
```

Comment:
This test is failing now:
```
=================================== FAILURES ===================================
____________________ test_tool[select_from_dataset_test_2] _____________________

testcases = ('select_from_dataset_test_2', <functional.test_toolbox.TestForTool_select_from_dataset testMethod=runTest>, 1)
driver = <functional.test_toolbox_pytest.DefaultGalaxyTestDriver object at 0x7fb17b8cc290>

    @pytest.mark.parametrize("testcases", cases(), ids=idfn)
    def test_tool(testcases, driver):
        test = testcases[1]
>       test.do_it(test_index=testcases[2])

test/functional/test_toolbox_pytest.py:84: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/functional/test_toolbox.py:35: in do_it
    verify_tool(tool_id, self.galaxy_interactor, resource_parameters=resource_parameters, test_index=test_index, tool_version=tool_version, register_job_data=register_job_data)
lib/galaxy/tool_util/verify/interactor.py:1041: in verify_tool
    raise e
lib/galaxy/tool_util/verify/interactor.py:1034: in verify_tool
    tool_response = galaxy_interactor.run_tool(testdef, test_history, resource_parameters=resource_parameters)
lib/galaxy/tool_util/verify/interactor.py:456: in run_tool
    submit_response_object = ensure_tool_run_response_okay(submit_response, "execute tool", inputs_tree)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

submit_response_object = <Response [400]>, request_desc = 'execute tool'
inputs = {'collection': {'id': '529fd61ab1c6cc36', 'src': 'hdca'}, 'multiple': {'id': '5a78730880510954', 'src': 'hda'}, 'select_collection': ['75', '58'], 'select_multiple': ['71', '62'], ...}

    def ensure_tool_run_response_okay(submit_response_object, request_desc, inputs=None):
        if submit_response_object.status_code != 200:
            message = None
            dynamic_param_error = False
            try:
                err_response = submit_response_object.json()
                if "param_errors" in err_response:
                    param_errors = err_response["param_errors"]
                    if "dbkey" in param_errors:
                        dbkey_err_obj = param_errors["dbkey"]
                        dbkey_val = dbkey_err_obj.get("parameter_value")
                        message = "Invalid dbkey specified [%s]" % dbkey_val
                    for value in param_errors.values():
                        if isinstance(value, dict) and value.get("is_dynamic"):
                            dynamic_param_error = True
                if message is None:
                    message = err_response.get("err_msg") or None
            except Exception:
                # invalid JSON content.
                pass
            if message is None:
                template = "Request to %s failed - invalid JSON content returned from Galaxy server [%s]"
                message = template % (request_desc, submit_response_object.text)
>           raise RunToolException(message, inputs, dynamic_param_error=dynamic_param_error)
E           galaxy.tool_util.verify.interactor.RunToolException: parameter 'select_single': an invalid option ('68') was selected (valid options: 4,1,5,7,8,10,9,2,6,3), parameter 'select_multiple': invalid options ('71,62') were selected (valid options: 4,1,5,7,8,10,9,2,6,3), parameter 'select_collection': invalid options ('58,75') were selected (valid options: 4,1,5,7,8,10,9,2,6,3)
```
Do you need to update `<column name="name" index="0"/>` and `<column name="value" index="0"/>` ?